### DISCUSSION TITLE
Confusion about cooldown and m_s paramter.

### DISCUSSION THREAD

**Discussion by pinouche** [esa/SpOC2#5] [Morphing Rovers Q&A] [answered]
I was wondering what the following sentence means in the description of the challenge:

If  m_s > 0 and the cooldown is 0, then the rover switches to the mode that is best for the current terrain and the cooldown is activated.

I have two questions:

1) It's unclear what the cooldown is (to me at least), as it has not really been discussed in the description. 
2) What is meant by best here? If I understand correctly, all 4 modes are used and the best one is chosen? 

Thanks!

**Comment by dodo47** [comment, marked as answer]
Hi pinouche!

Thanks for your questions :) 

1. Whenever the rover switches mode, it cannot initiate another mode switch for a fixed amount of time steps (i.e., 8 time steps here). That's what we call "cooldown" (borrowed from gaming, we indeed forgot to define the term in our description, sorry!). In the code, we keep track of this with a "cooldown" variable that simply counts down from 8 to 0 whenever the rover switches mode. If the variable has a value larger than 0, mode switching is not possible.

2. When switching, the rover will test all forms against the terrain it currently is standing on and choose the one that yields the highest velocity.

I hope that answers your questions :)

Best and good luck!
Dominik

**Reply by pinouche** [reply]
Thanks for the reply. That makes more sense now.
